android_kernel_tools.rc

```diff
@@ -673,20 +674,26 @@ function bootzip()
   if [ "${PushToDevice}" = 'true' ]; then
     echo '';
 
-    # Waiting for adb device
+    # Wait for device
     timeout 10 ${adb} 'wait-for-device';
 
-    # Push to common '/sdcard/' path
-    echo "  bootzip: Pushing to device /sdcard/${OutFile}";
-    timeout 10 ${adb} 'wait-for-device';
-    adb push "${TargetFile}" '/sdcard/';
+    # Check if '/sdcard' exists
+    if [ "$(${adb} shell 'test -d /sdcard/ 2> /dev/null; echo $?')" -eq '0' ]; then
+      adb_targetdir='/sdcard';
 
-    # If push failed, push to common '/data/media/0/' path
-    if [ -z "$(adb shell 'ls ${OutFile} 2> /dev/null')" ]; then
-      echo "           Failed pushing to /sdcard/${OutFile}";
-      echo "  bootzip: Pushing to device /data/media/0/${OutFile}";
-      adb push "${TargetFile}" '/data/media/0/';
-    fi;
+    # Check if '/data/media/0' exists
+    elif [ "$(${adb} shell 'test -d /data/media/0/ 2> /dev/null; echo $?')" -eq '0' ]; then
+      adb_targetdir='/data/media/0';
+    fi
+
+    # Check if device is connected
```

Comment:
Should not have this comment here or the waitfordevice below, because it's already done higher and device accessed for tests. Remove them and add a "# Push zip to device target"